skillbridge/src/component/CommonFunctionalities/FAQSection/FAQSection.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useReducer } from 'react';
+import { closeIcon, plusIcon } from '../../../assets/resource/iconResource';
+import { faqForwardButton } from '../../../assets/resource/imgResource';
+import { FAQList } from '../../Data/FAQData';
+
+
+type ActionType =
+  | { type: 'SHOW_ANSWER'; id: number }
+  | { type: 'HIDE_ANSWER' }
+  | { type: 'TOGGLE_VIEW_ALL' };
+
+
+//state type
+interface State {
+  showAnswer: number | null;
+  viewAll: boolean;
+}
+
+const faqReducer = (state: State, action: ActionType): State => {
+  switch (action.type) {
+    case 'SHOW_ANSWER':
+      return { ...state, showAnswer: action.id };
+    case 'HIDE_ANSWER':
+      return { ...state, showAnswer: null };
+    case 'TOGGLE_VIEW_ALL':
+      return { ...state, viewAll: !state.viewAll };
+    default:
+      return state;
+  }
+};
+
+function FAQSection() {
+  const initialState: State = {
+    showAnswer: null,
+    viewAll: false,
+  };
```

Comment:
Can't this be moved outside the component and explain why do you have to put this inside the component